MetabolicAtlas/MetabolicAtlas
# chore: add support for Matomo and Hotjar environment variables in docker compose files

**Related issue(s) and PR(s)**  
This PR closes https://github.com/MetabolicAtlas/private-issues/issues/122.



**Type of change**  

 
**List of changes made**  

This adapts the usage of environment variables to be more consistent with the recent rework of handling other environment specific changes.

Previously, the now removed `frontend/.env.production` file was used for both the dev and prod servers. Now it is possible to set different values for each remote server by editing the `env-*.env` files. The specific Matomo ID for each environment can be found in the related issue.

**Testing**  

- Instructions on how to test

It is most straightforward to test this by deploying to the dev server.
1. Edit your `env-dev.dev` file according to the updated `env-local.env.sample` file. Only add the `VUE_APP_MATOMOID` as `VUE_APP_HOTJAR` should only be added for the prod server.
2. Deploy to the dev server: `source proj.sh` followed by `deploy-stack dev`.
3. After the deployment, visit the dev server in a browser and open the Network tab in the developer tools pane.
4. Verify that the correct Matomo ID is set in the `matomo.php?...` request as shown in the screenshot. It helps to enter `matomo` in the Filter field to help find the request.

![image](https://user-images.githubusercontent.com/423498/151210979-b144ca81-7ae0-428b-8dd3-71517cee7f9a.png)


**Further comments**  

I'm not sure if the Matomo id is needed for the local environment. If not, we could discard the changes for the `docker-compose-local.yml` file. On the other hand, perhaps it is ok to leave it in as we can simplify omit setting the variable by not adding it the `env-local.env` file.

**Checklist**
